internal/driver/driver.go

```diff
@@ -303,18 +315,29 @@ func (d *Driver) createOnvifClient(deviceName string) error {
 	return nil
 }
 
-func (d *Driver) getCredentials(secretPath string) (config.Credentials, errors.EdgeX) {
-	credentials := config.Credentials{}
-	deviceService := sdk.RunningService()
+// tryGetCredentials will attempt one time to get the credentials located at secretPath from
+// secret provider and return them, otherwise return an error.
+func (d *Driver) tryGetCredentials(secretPath string) (config.Credentials, errors.EdgeX) {
+	secretData, err := sdk.RunningService().SecretProvider.GetSecret(secretPath, secret.UsernameKey, secret.PasswordKey)
+	if err != nil {
+		return config.Credentials{}, errors.NewCommonEdgeXWrapper(err)
+	}
+	return config.Credentials{
+		Username: secretData[secret.UsernameKey],
+		Password: secretData[secret.PasswordKey],
+	}, nil
+}
 
+// getCredentials will repeatedly try and get the credentials located at secretPath from
+// secret provider every CredentialsRetryTime seconds for a maximum of CredentialsRetryWait seconds.
+// Note that this function will block until either the credentials are found, or CredentialsRetryWait
+// seconds have elapsed.
+func (d *Driver) getCredentials(secretPath string) (credentials config.Credentials, err errors.EdgeX) {
```

Comment:
@cloudxxx8,  We tend to be doing this is all our device services. I am thinking this retry needs to be built into the SecretProvider. Something like GetSecretWithRetry(...). I will create issues.